lib/omni-sharp-server/client-manager.ts

```diff
@@ -12,11 +18,44 @@ class ClientManager {
     private _projectPaths: string[] = [];
     private _activated = false;
     private _temporaryClients: { [path: string]: RefCountDisposable } = {};
+    private _omnisharpAtom: typeof OmniSharpAtom;
+    private _activeClients: Client[] = [];
+
+    private _aggregateClient = new CompositeClient();
+    public get aggregateClient() { return this._aggregateClient; }
+
+    public _clientsSubject = new ReplaySubject<Client[]>(1);
+    private _clientStateObservable = this._clientsSubject
+        .flatMap(x => Observable.combineLatest(x.map(z => z.state), function(...values) { return values; }));
```

Comment:
Not big on comments but this could use one describing what it does and what its for :)